### PR TITLE
New client missing feats

### PR DESCRIPTION
**Pull request by ohadbitt**
# Removed
builder.maxConnectionsPerRoute as it was not used in new client

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java [hunk 1]
```diff
@@ -142,8 +153,13 @@ public static DataServiceException createExceptionFromResponse(String url, HttpR
                 isPermanent);
     }
 
+    private Context getContextTimeout(long timeoutMs) {
+        int requestTimeout = timeoutMs > Integer.MAX_VALUE ? Integer.MAX_VALUE : Math.toIntExact(timeoutMs) + CLIENT_SERVER_DELTA_IN_MILLISECS;
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java:157]: Maybe we should have a lower hard cap than MAX_VALUE, I think in the server side it's 1 hour

File: data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java [hunk 2]
```diff
@@ -142,8 +153,13 @@ public static DataServiceException createExceptionFromResponse(String url, HttpR
...
+        return Context.NONE.addData("azure-response-timeout", Duration.ofMillis(requestTimeout));
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java:158]: Is this documented somewhere? we might want to put a comment with the link for future reference

File: data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java [hunk 3]
```diff
@@ -15,12 +15,16 @@
...
+import java.io.UncheckedIOException;
 import java.lang.invoke.MethodHandles;
+import java.time.Duration;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 
 public abstract class BaseClient implements Client, StreamingClient {
 
     private static final int MAX_REDIRECT_COUNT = 1;
+    private static final int CLIENT_SERVER_DELTA_IN_MILLISECS = (int) TimeUnit.SECONDS.toMillis(30);
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java:27]: Maybe we should take this to change the name, EXTRA_TIMEOUT_FOR_CLIENT or something like that

File: data/src/main/java/com/microsoft/azure/kusto/data/exceptions/ExceptionUtils.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+
+public class ExceptionUtils {
+    public static DataServiceException createExceptionOnPost(Exception e, URL url, String kind) {
+        boolean permanent = false;
+        String prefix = "";
+        if (e instanceof IOException) {
```
Comment (AsafMah) [inline data/src/main/java/com/microsoft/azure/kusto/data/exceptions/ExceptionUtils.java:13]: I think if you already change the exception message, you might as well make a proper subclass for IODataServiceException , so the users could filter on them too

**Review by AsafMah — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Moved HTTP request building logic into a builder class.
 - [BREAKING] Redirects are disabled by default. Use ClientRequestProperties "client_max_redirect_count" option 
     to enable. Default changed to 0.
+- Removed  maxConnectionsPerRoute as it was not easily provided by azure-core.
```
Comment (AsafMah) [inline CHANGELOG.md:17]: Add Breaking tag? 
Also you have a double space

File: data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java (hunk 2 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/BaseClient.java:158]: not exactly but ill ref the changelog

**Review by AsafMah — APPROVED** [review]

**Review by AsafMah — APPROVED** [review]

**Pull request merged**